internal/cli/compile/compile.go

```diff
@@ -312,60 +314,69 @@ func runCompileCommand(cmd *cobra.Command, args []string, srv rpc.ArduinoCoreSer
 		}
 	}
 
+	successful := (compileError == nil)
 	profileOut := ""
-	if dumpProfile && compileError == nil {
-		// Output profile
-
-		libs := ""
-		hasVendoredLibs := false
-		for _, lib := range builderRes.GetUsedLibraries() {
-			if lib.GetLocation() != rpc.LibraryLocation_LIBRARY_LOCATION_USER && lib.GetLocation() != rpc.LibraryLocation_LIBRARY_LOCATION_UNMANAGED {
-				continue
+	stdIO := stdIORes()
+
+	if dumpProfile {
+		if successful {
+			// Output profile
```

Comment:
IMHO the profile dump should be moved to a separate function. It also seems to output a yaml format, couldn't be simple to use a yaml encoder? Anyway, those aren't related to the bugfix and could be addressed later.

Reply:
> the profile dump should be moved to a separate function

unfortunately, it's too late to change, maybe in a CLI v2

> couldn't be simple to use a yaml encoder?

probably yes, I don't remember why it has been made like this, but yes food for another PR.